src/lib/commands.rs

```diff
@@ -590,39 +594,42 @@ where
         agreed_shared_secret_id,
         KeySerialization::Raw
     ))
-    .map_err(|e| {
-        error!("Deserialization error: {:?}", e);
+    .map_err(|_e| {
+        error!("Deserialization error: {:?}", _e);
         Error::InternalError
     })?;
     syscall!(w.trussed.delete(agreed_shared_secret_id));
 
-    w.send_to_output(CommandOpenPGPDecryptResponse {
-        data: DataBytes::from_slice(&serialized_shared_secret.serialized_key)
-            .map_err(|_| Error::InternalError)?,
-    });
+    send_to_output(
+        CommandOpenPGPDecryptResponse {
+            data: DataBytes::from_slice(&serialized_shared_secret.serialized_key)
+                .map_err(|_| Error::InternalError)?,
+        },
+        reply,
+    );
 
     Ok(())
 }
 
 #[inline(never)]
-pub fn cmd_decrypt<C>(w: &mut Webcrypt<C>) -> CommandResult
+pub fn cmd_decrypt<C>(
+    w: &mut WebcryptInternal<C>,
+    req: CommandDecryptRequest,
+    reply: &mut Message,
+) -> CommandResult
 where
     C: WebcryptTrussedClient,
 {
-    let req = match w.get_input_deserialized() {
-        Ok(x) => Ok(x),
-        Err(e) => {
-            error!("Deserialization error: {:?}", e);
-            Err(e)
-        }
-    };
-
-    let req: CommandDecryptRequest = req.map_err(|_| Error::BadFormat)?;
     w.session
-        .check_token_res(req.tp.clone())
+        .check_token_res(&req.tp)
         .map_err(|_| Error::RequireAuthentication)?;
 
-    let (kh_key, mech, is_rk) = get_key_from_keyhandle(w, req.keyhandle.clone())?;
+    let (kh_key, mech, is_rk) = {
+        get_key_from_keyhandle(
+            w,
+            req.keyhandle,
+        )?
+    };
```

Comment:
Same here, why the brackets?

```suggestion
    let (kh_key, mech, is_rk) = get_key_from_keyhandle(w, req.keyhandle)?;
```

Reply:
Oopsie. That was left from an experiment, when I wanted to drop earlier the borrow. No need for them here of course.